### PR TITLE
Implement `div_ceil` for `NonZero<unsigned>`

### PR DESCRIPTION
**Pull request by tyilo**
ACP: https://github.com/rust-lang/libs-team/issues/471

### CODE REVIEW

File: library/core/src/num/nonzero.rs [hunk 1]
```diff
@@ -1214,6 +1214,35 @@ macro_rules! nonzero_integer_signedness_dependent_impls {
...
+            #[doc = concat!("let two = NonZero::new(2", stringify!($Int), ").unwrap();")]
+            #[doc = concat!("let three = NonZero::new(3", stringify!($Int), ").unwrap();")]
+            /// assert_eq!(three.div_ceil(two), two);
+            /// ```
+            #[unstable(feature = "unsigned_nonzero_div_ceil", issue = "none")]
+            #[must_use = "this returns the result of the operation, \
+                          without modifying the original"]
+            #[inline]
+            pub const fn div_ceil(self, rhs: Self) -> Self {
+                let v = self.get().div_ceil(rhs.get());
```
Comment (joboet) [inline library/core/src/num/nonzero.rs:1241]: Could you inline `div_ceil` here? That will lead to unchecked operations being used for the division and remainder which might allow some more optimizations.

Comment (tyilo) [reply library/core/src/num/nonzero.rs:1241]: Do you mean something like the following?

```rust
pub const fn div_ceil(self, rhs: Self) -> Self {
  let a = self.get();
  let b = rhs.get();
  let d = a / b;
  let r = a % b;
  let v = if r > 0 {
    d + 1
  } else {
    d
  };
  // SAFETY: ceiled division of two positive integers can never be zero.
  unsafe { Self::new_unchecked(v) }
}
```

Why would this allow more optimizations? They seem to produce the same ASM: https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=735568e0474e49fa0a1b86d773a486d0

Comment (joboet) [reply library/core/src/num/nonzero.rs:1241]: Ah right, LLVM can infer the non-zero aspect because of `get`. Nevermind...

**Comment by joboet** [comment]
Thank you
@bors r+ rollup

**Comment by joboet** [comment]
Yes, please. In fact, that should have been done before merging this, but I forgot about it, sorry!

**Pull request merged**